staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -390,7 +236,6 @@ func newETCD3Storage(c storagebackend.ConfigForResource, newFunc func() runtime.
 		once.Do(func() {
 			stopCompactor()
 			stopDBSizeMonitor()
-			client.Close()
```

Comment:
Is this 100% safe? For the kube-apiserver I can't think of any actual problems, but this staging repo allows anyone to write a binary that starts an API server, this appears to leak a client in that case..

Reply:
This `client` always comes from `newETCD3Client` which always comes from `TransportConfig.Client` which is always lifecycled with the `ctx` passed into `TransportConfig.Complete`.  Since the fields are private, it is impossible to get a client without calling `Complete` meaning any custom code that calls into this function would be at fault if it chose to pass in something like `context.TODO()`.  The standard wiring used by aggregated API servers always passes in a valid context.  Is there some other leak that you are seeing?

Reply:
Ignore me, I had missed the `close()` in `getClient()` :stuck_out_tongue: